geoprocessing/src/modules/planning-units/planning-units.job.ts

```diff
@@ -0,0 +1,72 @@
+import { Logger } from '@nestjs/common';
+import { Job } from 'bullmq';
+import { geoprocessingConnections } from 'src/ormconfig';
+import { createConnection } from 'typeorm';
+
+const logger = new Logger('planning-units-job-processor');
+
+export enum PlanningUnitGridShape {
+  square = 'square',
+  hexagon = 'hexagon',
+  fromShapefile = 'from_shapefile',
+}
+export interface PlanningUnitsJob {
+  countryId?: string;
+  adminRegionId?: string;
+  adminAreaLevel1Id?: string;
+  adminAreaLevel2Id?: string;
+  planningUnitGridShape: PlanningUnitGridShape;
+  planningUnitAreakm2: number;
+  extent?: Record<string, unknown>;
+}
+
+export default async (job: Job<PlanningUnitsJob> | any) => {
+  logger.debug(`Start planning-units processing for ${job.id}...`);
+  if (job.name === 'create-pu') {
+    const connection = await createConnection(geoprocessingConnections.default);
+    try {
+      let subquery: string;
+      const gridShape = {
+        square: 'ST_SquareGrid',
+        hexagon: 'ST_HexagonGrid',
+      };
+      if (job.data.extent) {
+        subquery = `(SELECT (${gridShape[job.data.planningUnitGridShape]}(${
+          Math.sqrt(job.data.planningUnitAreakm2) * 1000
+        },
+                            ST_Transform(ST_GeomFromGeoJSON('${JSON.stringify(
```

Comment:
ok I have tested and so far, if the users try to pass a negative value the DB on project creation throws an error (i love enforcing uint in the DB table) so from there we should be safe. but I had also add at the job level a check and an error trow just in case